text/0976-glint-as-default.md

```diff
@@ -0,0 +1,161 @@
+>
+
+> There are tradeoffs to choosing any path, please attempt to identify them here.
+>
+
+There are some potential drawbacks to adopting Glint:
+
+**Learning Curve:** Developers who are not familiar with TypeScript may face a learning curve when integrating Glint into their Ember projects.
+
+**Migration Effort:** Existing projects may require effort to migrate from not having a template checking system to Glint.
+
+**********************************************Registries Boilerplate:**********************************************
+
+Because Ember's template resolution occurs dynamically at runtime when using `.hbs` files, Glint needs a way of mapping the names used in your templates to the actual backing value they'll be resolved to. This takes the form of a "type registry" as defined above.
+
+Note that this is *not* an issue for users using `.gts` templates.
+
+## Alternatives
+
+> What other designs have been considered? What is the impact of not doing this?
+>
+
+> This section could also include prior art, that is, how other frameworks in the same domain have solved this problem.
+>
+
+The primary alternative to adopting Glint is to continue using Ember without a template type-checking system. However, this would mean missing out on the benefits of static type checking and improved tooling provided by Glint.
+
+## Unresolved questions
+
+> Optional, but suggested for first drafts. What parts of the design are still
+TBD?
+>
+
+### Is there redundancy between Ember Language Server and Glint Language Server?
+
+We should verify that there are no redundancies between the Glint Language Server and the Ember Language Server used in IDEs. If there is, we should answer the following questions:
+
+Should Glint Language Server features be merged into the Ember Language Server? Alternatively, should we remove the redundancy between them but keep them separate?
+
+### Does the Glint Language Server need full feature parity with the TypeScript Language Server before this is merged?
+
+Currently, we recommend Glint users enable the Glint Language Server in their IDE and disable the TypeScript Language Server to avoid duplicate errors. The Glint Language Server does not, however, currently have full feature parity with the TypeScript Language Server, so in doing so, users are losing some language server features. There is [work in progress](https://github.com/typed-ember/glint/issues/626) to rectify this issue.
+
+### Should we require that GTS is recommended before we recommend Glint?
+
+As noted above, using Glint with `.hbs` files, there is boilerplate necessary for Glint to recognize component invocations in the templates. With `.gts` files, this is not an issue.
+
+In the opinion of the authors of this RFC, the benefit of Glint outweighs the downsides of requiring the boilerplate.
+
+Additionally, `.gts` is not supported for route templates by default, so that issue would need to be resolved, delaying the Glint recommendation further.
```

Comment:
a note for readers, you can use gts in route templates via: https://github.com/discourse/ember-route-template

example: 
- https://github.com/NullVoxPopuli/limber/tree/main/apps/repl/app/templates